ops/testing.py

```diff
@@ -762,9 +764,14 @@ def _update_config(
         config = self._backend._config
         if key_values is not None:
             for key, value in key_values.items():
-                config[key] = value
+                if ignore_missing or key in self._backend._config:
+                    config[key] = value
+                else:
+                    raise ValueError("unknown config option: '{}'".format(key))
         for key in unset:
-            config.pop(key, None)
+            # For now, unsetting will just reset the config item to 'None', which does not respect
+            # the 'default' key.
+            config[key] = None
```

Comment:
This probably is not great for charm authors, since 'juju config --reset foo app' will actually set foo back to the default.
Maybe we should be splitting the config into a 'defaults' separate from the active value? And then use:
`get_config(key) val = config.get(key, None); if val is None: val = default_config.get(key, None)`
?

Reply:
Fixed now